Forte.EpiServer.AzureSearch/Query/Extensions/FilterByAccessExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using System.Linq;
+using EPiServer.Security;
+using Forte.EpiServer.AzureSearch.Model;
+
+namespace Forte.EpiServer.AzureSearch.Query.Extensions
+{
+    public static class FilterByAccessExtensions
+    {
+        /// <summary>
+        /// Method filters search items by allowed roles and users
+        /// </summary>
+        /// <param name="queryBuilder"></param>
+        /// <returns></returns>
+        public static AzureSearchQueryBuilder FilterOnReadAccess(this AzureSearchQueryBuilder queryBuilder)
+        {
+            var roles = PrincipalInfo.Current?.RoleList ?? Enumerable.Empty<string>();
+            var user = PrincipalInfo.Current?.Name ?? string.Empty;
+            
+            var userAccessFilter = new AzureSearchQueryFilter(nameof(ContentDocument.AccessUsers), ComparisonExpression.Eq, user)
+            {
+                GroupingExpression = GroupingExpression.Any
+            };
+            
+            var accessFilters = roles
```

Comment:
Maybe roles filter should be extracted to variable and then concatenated with userAccessFilters?